packages/runtime/container-runtime/src/summaryManager.ts

```diff
@@ -80,65 +99,46 @@ export class SummaryManager extends EventEmitter implements IDisposable {
             "SummaryManager",
             {all:{ clientId: () => this.latestClientId }});
 
-        this.connected = context.connected;
-        if (this.connected) {
-            this.setClientId(context.clientId);
-        }
+        this.connectedState.on("connected", this.handleConnected);
+        this.connectedState.on("disconnected", this.handleDisconnected);
+        this.latestClientId = this.connectedState.clientId;
 
         // Track ops until first (write) connect
         const opsUntilFirstConnectHandler = (clientId: string, details: ISequencedClient) => {
-            if (this.opsUntilFirstConnect === -1 && clientId === this.clientId) {
+            if (this.opsUntilFirstConnect === -1 && clientId === this.connectedState.clientId) {
```

Comment:
given the notes on latestClientId, should this be using that instead of `connectedState.clientId`?

Reply:
When we know we're connected I use `connectedState` since it's the "source of truth".  I'm also thinking about lifting this delay logic out potentially since it also references `IContainerContext` and we can probably just delay the `start()` call from the `ContainerRuntime` instead.